modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -0,0 +1,753 @@
+          orientation = 90.;
+        }
+        else if (j == _pattern[i].size() - 1 && i == _pattern.size() - 1)
+        {
+          rotation_angle = -90.;
+          mesh_type = CORNER_MESH;
+          orientation = 180.;
+        }
+        else if (i == 0)
+        {
+          rotation_angle = 0.;
+          mesh_type = BOUNDARY_MESH;
+          orientation = 0.;
+        }
+        else if (i == _pattern.size() - 1)
+        {
+          rotation_angle = 180.;
+          mesh_type = BOUNDARY_MESH;
+          orientation = -180.;
+        }
+        else if (j == 0)
+        {
+          rotation_angle = 90.;
+          mesh_type = BOUNDARY_MESH;
+          orientation = -90.;
+        }
+        else if (j == _pattern[i].size() - 1)
+        {
+          rotation_angle = -90.;
+          mesh_type = BOUNDARY_MESH;
+          orientation = 90;
+        }
+        else
+          on_periphery = false;
+
+        if (on_periphery)
+        {
+          auto tmp_peripheral_mesh = dynamic_pointer_cast<ReplicatedMesh>(pattern_mesh.clone());
+          addPeripheralMeshRect(*tmp_peripheral_mesh,
+                                pattern,
+                                pitch_array.front(),
+                                extra_dist,
+                                num_sectors_per_side_array,
+                                peripheral_duct_intervals,
+                                rotation_angle,
+                                mesh_type,
+                                _create_interface_boundaries);
+
+          if (extra_dist_shift != 0)
+            cutOffPolyDeform(
+                *tmp_peripheral_mesh, orientation, y_max_0, y_max_n, y_min, mesh_type, 90.0);
+
+          if (i == 0 && j == 0)
+            out_mesh = std::move(tmp_peripheral_mesh);
+          else
+          {
+            // Retrieve subdomain name map from the mesh to be stitched and insert it to the main
+            // subdomain map
+            const auto & increment_subdomain_map = tmp_peripheral_mesh->get_subdomain_name_map();
+            out_mesh->set_subdomain_name_map().insert(increment_subdomain_map.begin(),
+                                                      increment_subdomain_map.end());
+
+            MeshTools::Modification::translate(
+                *tmp_peripheral_mesh, deltax + j * input_pitch, deltay, 0);
+            out_mesh->stitch_meshes(*tmp_peripheral_mesh,
+                                    OUTER_SIDESET_ID,
+                                    OUTER_SIDESET_ID,
+                                    TOLERANCE,
+                                    /*clear_stitched_boundary_ids=*/true);
+          }
+
+          continue;
+        }
+      }
+
+      if (_assign_control_drum_id && _pattern_boundary == "none" && _generate_core_metadata)
+        pattern_mesh.add_elem_integer(
+            "control_drum_id",
+            true,
+            is_control_drum_array[pattern] ? control_drum_azimuthals.size() : 0);
```

Comment:
so the default value for the id is the number of drums
and that gets overwritten right? why set this value?

Reply:
The value is 0 for non CD and `control_drum_azimuthals.size()` for CD. The `control_drum_azimuthals` increases its size by 1 everytime a CD is involved in the loop, this is actually the index instead of a const value.